RecoLocalCalo/EcalRecAlgos/src/KernelHelpers.cu

```diff
@@ -59,18 +225,157 @@ namespace ecal {
             5950, 5992, 6034, 6076, 6118, 6160, 6202, 6244, 6281, 6318, 6355, 6392, 6429, 6466, 6503, 6540, 6577,
             6614, 6649, 6684, 6719, 6754, 6784, 6814, 6844, 6874, 6904, 6934, 6964, 6994, 7024, 7054, 7079, 7104,
             7129, 7154, 7179, 7204, 7219, 7234, 7249, 7264, 7274, 7284, 7294, 7304, 7314};
-
+            
+            
+            __device__ int quadrant(int iX, int iY) {
+              bool near = iX >= 11;
+              bool far = !near;
+              bool top = iY >= 11;
+              bool bot = !top;
+              
+              int iquad = 0;
+              if (near && top)
+                iquad = 1;
+              if (far && top)
+                iquad = 2;
+              if (far && bot)
+                iquad = 3;
+              if (near && bot)
+                iquad = 4;
+              
+              return iquad;
+            }
+            
+            __device__ int sector(int iX, int iY) {
+              //  Y (towards the surface)
+              //  T
+              //  |
+              //  |
+              //  |
+              //  o---------| X  (towards center of LHC)
+              //
+              static const int idx_[] = {
+                // 1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20
```

Comment:
looks like the formatting got messed up ?

Reply:
same comment as before about a new PR with cpu and gpu  improvements in parallel

https://github.com/cms-sw/cmssw/blob/2c91040e71a5447c0e161e80ce914260713b8317/CalibCalorimetry/EcalLaserAnalyzer/src/MEEEGeom.cc#L42

(for reference about where to fix on cpu)